sdk/mobile-center/src/main/java/com/microsoft/azure/mobile/channel/DefaultChannel.java

```diff
@@ -110,6 +110,12 @@
     private Device mDevice;
 
     /**
+     * State checker. If this counter changes during a call to persistence, we have to ignore the result in the callback.
+     * Cancelling a database call would be unreliable, and if it's too fast you could still have the callback being called.
+     */
+    private int mStateChangeCounter;
```

Comment:
They say in Javadoc that its not a general replacement for int, and it has overhead and we still need synchronized for general thread safety (even before this patch the code was meant to be thread safe).